gateway/README.md

```diff
@@ -1 +1,15 @@
-Functions in this package implement supported XMLRPC methods/namespaces, and they are normally called by Controllers.
\ No newline at end of file
+This package contains the bussines rules of the application, and the implementation of all the usecases.
+All the other components of the application are plugins to this package.
+It should not contain any external dependency at all, and it should interact with the external packages only through interfaces.
+
+At the moment, the following use cases are implemented:
+
+- "Login": login to the Hub server, using credentials provided by the user. 
+- "Login with auth-relay mode": Extends the "Login" use case, and stores the credentials for further usage to login to the Uyuni peripheral servers.
+- "Login with Autoconnect mode": Extends the "Login with auth-realy mode" and includes the "Attach to servers" usecase.
+- "Attach to servers": login to the Uyuni peripheral servers of a have architecture.
+- "Logout": Logout from the Hub server and the Uyuni peripheral servers.
+- "Unicast": Execute a unicast call to a peripheral Uyuni server of the Hub architecture.
+- "Multicast": Execute a multicast call to many peripheral Uyuni servers of the Hub architecture.
+- "Proxy call to Hub": Redirect a call to the underlying regular Uyuni XMLPRC API of the Hub server.
+- "List user server IDs": Retrieve a the list of the server IDs of those Uyuni peripheral servers a user has access to.
```

Comment:
I see a risk enumerating use cases here, as they could fall out-of-sync with the code if we do not remember to change them here as well.

Could it be an alternative to move the content here in individual files implementing use cases in the form of comments, where they are more easily noticed in case of changes? Do you see any cons?

Reply:
Yes, that's an alternative.

To be honest, I would prefer to have it both ways, a "usecases" file and a comment a in the file implementing the use case, and have the discipline to keep this file synced (the same as having the discipline to add testing for the usecase, for example).

The purpose of "documenting" the use cases, in a super light way was to have them some where, so eventually the next guy who picks this up can have a better picture of the whole app.